---
title: "Bulk RNAseq Differential Expression Analysis"
author: "Mindy Hoang Tran"
date: "03/23/2025"

output:
  html_document:
    toc: true
    fig_caption: true
---
```{r echo = TRUE, results = 'hide', message = FALSE, warning = FALSE}
setwd("/projectnb/bf528/students/mindyt5/project-1-mindy-tran")

# Install fgsea if not already installed
if (!requireNamespace("fgsea", quietly = TRUE)) {
    install.packages("BiocManager")
    BiocManager::install("fgsea")
}


# Load necessary libraries
library(fgsea)
library(msigdbr)
library(DESeq2)
library(ggplot2)
library(ggrepel)
library(dplyr)
library(tibble)
library(pheatmap)
library(fgsea)
library(tidyverse)
library(patchwork)

```

## Evaluate the QC metrics for the full data

The sequencing reads show high overall quality, with median Phred scores consistently above Q30 across the read length. While the first 10 base pairs show slightly lower quality scores (ranging from Q26 to Q35), the scores stabilize above Q30 beyond first 10 base pairs and maintain high accuracy even at 100 base pairs. The per-sequence quality distribution confirms that large proportion of reads have high-quality scores, supporting the reliability of our sequencing data. The per-base sequence content suggests some nucleotide composition bias within the first 10 base pairs, but this diminishes by position 12 and the nucleotide distributions remain stable. The GC content distribution follows a normal pattern, suggesting no significant deviations due to contamination or library preparation bias. Adapter content is small, with less than 1% cumulative adapter presence detected, indicating that adapter trimming is likely unnecessary. Overall, our sequencing data is high quality has no major concerns that would impact downstream analysis.

## Filtering the counts matrix
I filtered out genes with less than 10 counts across all samples. Before filtering, the dataset contained 63,241 genes. After filtering, 27,723 genes remained. The distribution of gene counts before (red) and after (cyan) filtering is shown in the histogram below and shows the removal of low-expression genes.

```{r echo = TRUE, warning = FALSE}


# Read the counts matrix
counts <- read.table("results/concat_counts_matrix.tsv", header = TRUE, sep = "\t", row.names = 1, check.names = FALSE)

# Check the number of genes before filtering
num_genes_before <- nrow(counts)

# Apply filtering: remove genes that have less than 10 counts across all samples
filtered_counts <- counts[rowSums(counts) >= 10, ]

# Check the number of genes after filtering
num_genes_after <- nrow(filtered_counts)

# Report the number of genes before and after filtering
cat("Number of genes before filtering:", num_genes_before, "\n")
cat("Number of genes after filtering:", num_genes_after, "\n")

# Plot distribution of gene counts before and after filtering
before_counts <- rowSums(counts)
after_counts <- rowSums(filtered_counts)

df <- data.frame(
  Counts = c(before_counts, after_counts),
  Status = rep(c("Before Filtering", "After Filtering"), c(length(before_counts), length(after_counts)))
)
df$Status <- factor(df$Status, levels = c("Before Filtering", "After Filtering")) # make red in the back and cyan in the front

ggplot(df, aes(x = Counts, fill = Status)) +
  geom_histogram(bins = 50, alpha = 0.5, position = "identity") +
  scale_x_log10() + 
  theme_minimal() +
  labs(title = "Distribution of Gene Counts Before and After Filtering",
       subtitle = paste0("Number of genes before filtering: ", num_genes_before, "\n", "Number of genes after filtering: ", num_genes_after),
       x = "Log10(Counts)", y = "Number of Genes") +
  theme(legend.position = "top")
  

```




## Performing differential expression analysis using the filtered counts
Number of significant genes at padj < 0.001: 405

```{r echo = TRUE, warning = FALSE}
# DESeq2 requires a counts matrix, column data (sample information), and a formula
# the counts matrix *must be raw counts*
# Ensure it's a matrix and preserve row names
padj_thresh <- 0.001
count_mat <- as.matrix(filtered_counts)

coldata <- data.frame(
  condition = c("control", "experimental", "control", "experimental", "experimental", "control"),
  row.names = colnames(count_mat)  # Ensure row names match count matrix column names
)



# Create DESeq2 dataset
dds <- DESeqDataSetFromMatrix(countData = count_mat, 
                              colData = coldata, 
                              design = ~condition)

# Run DESeq2 normalization and DE analysis
dds <- DESeq(dds)

# Extract results
res <- results(dds, alpha = padj_thresh)  # Adjust threshold as needed

###### Significant DEGs ######
gene_names <- read.table('results/gencode_v45_gene_mapping.txt', header = TRUE, sep = "\t")

# Filter for significant genes
sig_genes <- res %>%
  as.data.frame() %>%
  filter(padj < padj_thresh) %>%
  rownames_to_column(var = "gene") %>% 
  arrange(padj)  %>% # genes ranked by padj
  left_join(gene_names, by=c('gene' = 'Ensembl_ID')) %>%
  relocate(Gene_Name, .after = gene)


# Choose an appropriate padj threshold and report the number of significant genes at this threshold.
# Report number of significant genes
num_sig_genes <- nrow(sig_genes)
print(paste0("Number of significant genes at padj < ", padj_thresh, ": ", num_sig_genes))

# A table containing the DESeq2 results for the top ten significant genes
top10_genes <- sig_genes %>%
  head(10)
top10_genes

```


### The results from a DAVID or ENRICHR analysis on the significant genes at your chosen padj threshold. Comment in a notebook what results you find most interesting from this analysis.

Many of the significant genes were related to cell development like cell adhesion (4.4E-7 pvalue), extracellular matrix (ECM) (3.7E-10 pvalue), and glycoprotein (3.7E-10 pvalue). This could mean the genes are important in how cells interact with their environment with tissue remodeling, metastasis, or immune responses to disease or treatments.

### RNAseq Quality Control Plots
The PCA plot indicates that the control and experimental samples have clear clustering and suggests that the experimental condition significantly influences gene expression. There is an experimental outlier that is far from all of the other samples, and could be due to batch effects or other technical issues. PC1 and PC2 also make up a large percentage of variance (86% and 10% respectively) which shows that they capture the major differences between the two conditions. 

Looking at the heatmap, the replicates are clustering together except for the one outlier I noticed in the PCA plot. The experimental outlier clusters with the control samples, so we'd need to check for possible mislabeling, contamination, or batch effects. Overall, this analysis indicates that the data is has clear experimental effects and mostly consistent sample relationships.

```{r}

# Perform variance-stabilizing transformation
vsd <- vst(dds, blind=TRUE)

# Extract the normalized counts
normalized_counts <- assay(vsd)

# PCA plot of PC1 vs PC2
pca_data <- plotPCA(vsd, intgroup="condition", returnData=TRUE)
percentVar <- round(100 * attr(pca_data, "percentVar"))

ggplot(pca_data, aes(PC1, PC2, color=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  theme_minimal() +
  ggtitle("PCA of Samples")

```




```{r}
######## Heatmap

# Compute Euclidean distance between samples
sample_dist <- dist(t(assay(vsd)))

# Convert to matrix for heatmap visualization
sample_dist_matrix <- as.matrix(sample_dist)
rownames(sample_dist_matrix) <- colnames(vsd)
colnames(sample_dist_matrix) <- colnames(vsd)

# Generate the heatmap
pheatmap(sample_dist_matrix, clustering_distance_rows=sample_dist, clustering_distance_cols=sample_dist,
         main="Sample-to-Sample Distance Matrix")

######### 
sampleDists <- dist(t(assay(vsd)))

library("RColorBrewer")
library('pheatmap')
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$condition, vsd$type, sep="-") # format sample-replicate names
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)

pheatmap(sampleDistMatrix,
         main="Version 2: Sample-to-Sample Distance Matrix",
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
```






### FGSEA Analysis
I am seeing pathways related to breast cancer (HUPER_BREAST_BASAL_VS_LUMINAL_UP, SCHUETZ_BREAST_CANCER_DUCTAL_INVASIVE_UP) and the extracellular matrix (REACTOME_ASSEMBLY_OF_COLLAGEN_FIBRILS_AND_OTHER_MULTIMERIC_STRUCTURES, REACTOME_EXTRACELLULAR_MATRIX_ORGANIZATION). This makes me wonder if this analysis is hinting at something in breast cancer cells and the tumor progression.


```{r echo = TRUE, warning = FALSE}
# Load DESeq2 results
res <- results(dds, contrast = c("condition", "experimental", "control"))
labeled_results <- rownames_to_column(res %>% as.data.frame(), 'Ensembl_ID')
id2gene_path <- '/projectnb/bf528/students/mindyt5/project-1-mindy-tran/results/gencode_v45_gene_mapping.txt'
gmt_file_path <- '/projectnb/bf528/students/mindyt5/project-1-mindy-tran/refs/c2.all.v2024.1.Hs.symbols.gmt'
min_size <- 5
max_size <- 500

make_ranked_log2fc <- function(labeled_results, id2gene_path) {
  #first load in the id2gene.txt appropriately
  id2gene <- read_delim(id2gene_path, col_names = c('Ensembl_ID', 'Gene_Name'))
  
  # join to add a new column in your labeled results that matches IDs to symbols
  labeled_results <- labeled_results %>% 
    left_join(id2gene, by='Ensembl_ID') %>%
    arrange(desc(log2FoldChange)) %>% # log2FC values in descending order
    drop_na()
  
  # generate a named vector of symbols
  result <- labeled_results$log2FoldChange
  names(result) <- labeled_results$Gene_Name
  
  return(result)
}


run_fgsea <- function(gmt_file_path, rnk_list, min_size, max_size) {
  # load GMT file
  pathways <- fgsea::gmtPathways(gmt_file_path)
  
  # Run fgsea and make tibble of results
  result <- fgsea::fgsea(
    pathways = pathways,
    stats = rnk_list,
    minSize = min_size,
    maxSize = max_size
  )
  result <- as_tibble(result)
  
  return(result)
}

top_pathways <- function(fgsea_results, num_paths){
  # get top n pathways ranked by positive and negative NES
  top_positive <- fgsea_results %>%
    arrange(desc(NES)) %>%
    head(num_paths) %>%  # Get top positive pathways
    mutate(top_NES = "pos")  # Add category
  
  top_negative <- fgsea_results %>%
    arrange(NES) %>%
    head(num_paths) %>%  # Get top negative pathways
    mutate(top_NES = "neg")  # Add category
  
  # Combine top positive and negative results
  top_results <- bind_rows(top_positive, top_negative)
  top_results$pathway <- gsub("_"," ", top_results$pathway)
  
  
  # Create the bar plot
  g <- ggplot(top_results, aes(x = reorder(stringr::str_wrap(pathway, 70), NES), y = NES, fill = top_NES)) +
    geom_bar(stat = "identity", show.legend = FALSE) +
    coord_flip() + # flip coordinates
    labs(title = "fgsea results for Hallmark MSigDB gene set",
         x = NULL,  
         y = "Normalized Enrichment Score (NES)",
         fill = "NES Category") +
    theme_minimal() +
    scale_fill_manual(values = c("pos" = "lightblue", "neg" = "pink")) +
    theme(
      axis.text = element_text(size = 6)
    )
  print(top_results$pathway)
  return(g)
}



# Main
ranked_list <- make_ranked_log2fc(labeled_results, id2gene_path)
fgsea_results <- run_fgsea(gmt_file_path, ranked_list, min_size, max_size)

padj_thresh <- 0.01
top_fgsea <- fgsea_results %>%
  filter(padj < padj_thresh) %>%
  arrange(desc(NES))  # Sort by enrichment score

# plot top 10 pathways
top_pathways(top_fgsea, num_paths = 10)

```



### Original Publication
Reproduce figures 3C and 3F from:
Chandra, V., Ibrahim, H., Halliez, C. et al. The type 1 diabetes gene TYK2 regulates β-cell development and its responses to interferon-α. Nat Commun 13, 6363 (2022). https://doi.org/10.1038/s41467-022-34069-z

#### Fig. 3c
##### Compare how many significant genes are up- and down-regulated in their findings in the paper and mine:

The paper had 319 upregulated and 412 downregulated genes (Fig. 3c, FDR<0.01). With my padj threshold of 0.01, I got 312 upregulated and 400 downregulated genes. 
```{r echo = TRUE, warning = FALSE}
# Figure 3c: Volcano plot

volcano_data <- labeled_results
volcano_data$diffexpressed <- "NO"

# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
volcano_data$diffexpressed[volcano_data$log2FoldChange > 0.2 & volcano_data$padj < padj_thresh] <- "UP"
# if log2Foldchange < -0.6 and pvalue < 0.05, set as "DOWN"
volcano_data$diffexpressed[volcano_data$log2FoldChange < -0.2 & volcano_data$padj < padj_thresh] <- "DOWN"

gene_mapping <- read.table('results/gencode_v45_gene_mapping.txt', header=T) %>% as_tibble()

volcano_data <- volcano_data %>% left_join(gene_mapping)
volcano_data$delabel[volcano_data$diffexpressed != "NO"] <- volcano_data$Gene_Name[volcano_data$diffexpressed != "NO"]


# label only top 5 genes up and down regulated
# Select top 5 most upregulated genes
top_up <- volcano_data %>%
  filter(diffexpressed == "UP") %>%
  arrange(pvalue) %>%
  head(20)

# Select top 5 most downregulated genes
top_down <- volcano_data %>%
  filter(diffexpressed == "DOWN") %>%
  arrange(pvalue) %>%
  head(20)

# Combine the selected genes
top_genes <- bind_rows(top_up, top_down)

# Plot volcano with selective labeling
ggplot(volcano_data, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed)) +
  geom_point() +
  geom_text_repel(data = top_genes, aes(label=Gene_Name), size = 3, max.overlaps = 30) + 
  theme_minimal() +
  scale_color_manual(values=c("blue", "darkgrey", "red"))+
  xlim(-10, 10) +
  labs(
    title = 'Figure 3c',
    subtitle = paste0('Number Upregulated: ', nrow(volcano_data %>% filter(diffexpressed == 'UP')), '\n', 
                      'Number Downregulated: ', nrow(volcano_data %>% filter(diffexpressed == 'DOWN')))
  )

```




#### Fig. 3f
##### Compare their enrichment results with your DAVID and GSEA analysis. Comment on any differences you observe and why there are discrepancies

My FGSEA analysis barplot also highlighted "Extracellular Matrix Organization" and "Neuronal System," but included additional pathways not from Reactome. The paper's Figure 3F focused solely on Reactome pathways, which may be why my padj values in my recreation of 3f below look so different.  My percentages also look very different from the paper. In my recreation of Figure 3C, I had 312 upregulated and 400 downregulated genes, while the paper reported 319 upregulated and 412 downregulated genes. These differences in Figure 3F below may come from differences in the total number of up- and downregulated genes detected.


```{r}


pathways <- c('Extracellular Matrix Organization', 'Integrin cell surface interactions', 'ECM proteoglycans', 'Signaling by Receptor Tyrosine Kinases', 'Regulation of beta cell development', 'Neuronal System', 'Regulation of gene expression in beta cells')

# Preprocess pathways
formatted_pathways <- toupper(gsub(" ", "_", pathways))
formatted_pathways <- paste0("REACTOME_", formatted_pathways)
de_genes <- volcano_data %>% filter(diffexpressed != 'NO')
de_genes <- de_genes$Gene_Name

# Compute percentage of DE genes per pathway
fgsea_filtered <- fgsea_results %>%
  mutate(clean_pathway = gsub("_", " ", sub("REACTOME_", "", pathway))) %>%
  filter(pathway %in% formatted_pathways) %>%
  mutate(
    percent_DE = (sapply(leadingEdge, length) / size) * 100,  # Compute percentage
    regulation = factor(ifelse(ES >= 0, "Upregulated", "Downregulated")) # Label regulation
  )




# Plot
p1 <- ggplot(fgsea_filtered %>% filter(regulation == 'Upregulated'), aes(
    x = percent_DE, 
    y = clean_pathway,
    fill = padj
  )) +
  geom_bar(stat = "identity") +  # Stacked bar plot
  labs(
    title = "Percentage of DE Genes in Pathways",
    x = "Percentage of DE Genes in Pathway (%)",
    y = "Pathway",
    fill = "Padj"
  ) +
  xlim(0, 50) + 
  theme_minimal() +
  theme(axis.text.y = element_text(size = 8))   +
  scale_fill_gradient(high = "blue", low = "red", 
                      limits = c(0.001, 0.05), 
                      oob = scales::squish)

p2 <- ggplot(fgsea_filtered %>% filter(regulation == 'Downregulated'), aes(
    x = percent_DE, 
    y = clean_pathway,
    fill = padj
  )) +
  geom_bar(stat = "identity") +  # Stacked bar plot
  labs(
    x = "Percentage of DE Genes in Pathway (%)",
    y = "Pathway",
    fill = "Padj"
  ) +
  xlim(0, 50) + 
  theme_minimal() +
  theme(axis.text.y = element_text(size = 8)) +
  scale_fill_gradient(high = "blue", low = "red", 
                      limits = c(0.001, 0.05), 
                      oob = scales::squish)

p1/p2 + plot_layout(ncol = 1, heights = c(1,1))

```




# Project 1 - Discussion Questions
These questions are meant to help me assess how well I’ve conveyed some of the
material as well as challenge you to think conceptually about some of the
underlying science. Some of the questions have a definitive "right" answer while
some of them are more open-ended and I'm just looking for your thought process
and interpretation. All of these questions can be answered in a few sentences so
do your best to be concise.

Please copy these questions into your single Rmd or notebook that you've been 
working on your DE analysis. All of your analyses, writing, and answers to these
questions should be in a single notebook. 


## RNAseq

**1. List the major high-level steps of a basic RNAseq experiment to look for differentially expressed genes. At each step, list what data you need to perform each step and what format they are in (if applicable). At minimum, there are 4 essential steps.**


First, I aligned reads to the genome with the fastq.gz file for each sample, and with STAR are converted into BAM files. Second, I generated gene-level counts with the BAM files from STAR to quantify the expression of genes, and then I concatonated all the counts together in a large matrix. Third, I filtered out lowly expressed genes from the concatenated counts matrix (tsv in this case) to reduce noise. Finally, I performed differential expression analysis with DESeq2 with the filtered counts matrix (which also performs normalization) to identify genes that are differentially expressed across the conditions.


**2. Consider the following FastQC plot.**

```{r}
knitr::include_graphics("fastqc_plot.png")
```


**2a. What aspect of the data does this plot show?**
It shows the GC content distribution across all reads and helps us identify contamination or overrepresented sequences.


**2b. Make an interpretation of this plot assuming the data type was RNASeq.**
There is some overrepresented sequence with the peaks at x=75-95, indicating contamination from another organism or rRNA since the peak is fairly broad. I would also say there is an over-represented sequence since there is a sharper peak on the normal distribution at x=39.
- We have many reads in the high forties but many in the high GC content
- they could be adapter sequences that still needs to be trimmed off
- rRNA that has not been depleted
- overall, there are two distinct populations with very different GC content



**2c. Do you think this plot indicates there was a problem with the dataset?**
**Explain your answer.**
Yes, I think there is a problem because I see a wide, broad peak around x=75-95, it could indicate contamination from another organism. There is also a shifted GC content peak from the normal, expected distribution, so there may be rRNA contamination as well.




**2d. Make a hypothesis about what the problem represents. Outline a bioinformatics strategy that would enable you to confirm your hypothesis.**
Hypothesis: The broad peak around x=75-95 could represent contamination from rRNA or another organism, and the shifted peak around x=39 could further indicate rRNA contamination.

Outline:
1. Align the raw reads to a rRNA reference sequence (like with Bowtie2) to quantify the amount of rRNA contamination.
2. Use BLAST to look at unmapped reads that could be from bacteria, viral, or other unexpected contaminants.
3. After filtering out contaminating sequences, look at the GC content distribution again to see if the peak at x=39 changes to a more expected pattern.


**3. What is a splice-aware aligner? When is it important to use a splice-aware aligner?**
Splice aware aligners use knowledge of splice junctions and exon/intron structure to better map reads. It's important to always use a splice-aware aligner for RNAseq against a reference genome because we want more accurate mapping. Genes encoded in DNA have exons and introns, and with added complexity from alternative splicing, contigs can be from different exons spliced together. Splice-aware
aligners help us map more accurately.




**4. What does a “gene-level” count as produced by VERSE or any other counting tool in a RNAseq experiment represent?**

The gene-level counts give a general quantification of gene expression rather than counting individual transcripts. It's the total number of reads mapped to a gene where the reads align to any of a gene's exons. Formally, it is the sum of the union of all the exon accounts.




**5. In your own words, briefly describe what information the matching GTF for a reference genome stores.**
GTF files holds gene features and encodes relationships and contains things like the gene location, transcript, and exon locations, gene names, transcript IDs, and coding sequences.





**6. When counting alignments using VERSE or any other utility, why do we need to provide the matching reference genome GTF file?**
We use the GTF file to quantify the gene expression, and we need it to assign reads to genes.The gene feature information in the GTF file helps us properly count the genes present in our samples.






**7. Let’s pretend that this was a GSEA result from an experiment where we treated 293T cells with a drug and compared changes in gene expression to wild-type cells treated with a vehicle control. The differential expression results are relative to the control cells (i.e. a positive fold change means a gene is upregulated upon treatment with the drug)**

**Assume the following result is statistically significant with a positive NES (normalized enrichment score) and that it represents a GSEA experiment performed on the entire list of genes discovered in the experiment ranked by log2FoldChange (i.e. genes that are “upregulated” in the cells treated with drug are on the “left” and genes that are “downregulated” due to treatment are on the “right”).**

```{r}
knitr::include_graphics("gsea_plot.png")
```


**7a. Form a valid interpretation / conclusion of the results shown in the plot given the above setup.**
The gene set genes have a high degree of correlation with the disease phenotype since most of the ES was gained from the left part of the plot. When no gene set genes were encountered for a long time, and we moved from the leading edge, the ES decreased quickly. 

Since the ES reaches its peak on the left side of the plot, I think that the gene set is enriched in the upregulated (positively correlated) genes in drug-treated cells. This suggests that the pathway associated with this gene set might be activated by the drug treatment.


**7b. Now consider that all of the genes driving the enrichment are all activators of the inflammatory pathway. Does your interpretation change and if so, how?**

If they are all activators of the inflammatory pathway, then I'll add onto my interpretation and say that the drug treatment may induce an inflammatory response. I think this suggests the drug could trigger pathways for immune activation or stress response.



**7c. Finally, consider that all of the genes driving the enrichment all function to inhibit inflammation. Does your interpretation change and if so, how?**
If they all inhibit inflammation, then my interpretation does change. I would say that the drug suppresses inflammatory signaling in our cells, and the drug could be an anti-inflammatory drug.





**8. Rank the following quality control metrics for a 2x100nt paired-end illumina  mRNAseq dataset from most concerning to least concerning. Provide a brief statement on where you ranked each and why. Assume that the data has not been processed and was provided as-is from the sequencing machine. **

- Unequal Read Lengths
- Average PHRED score < 20 in the last 10 bases
- 15% of reads have identical sequences
- 50% of reads are multimapped after alignment to the appropriate genome
- 10% of reads are unmapped after alignment to the appropriate genome
- Non-random nucleotide distribution in the first 6 bases
- Nucleotide frequencies of ACTG are not equal over the entire read
- Unequal number of forward and reverse reads

*1. 50% of reads are multimapped after alignment to the appropriate genome*

I am the most concerned about this because a large portion of the reads align to multiple locations in the genome, and this indicates issues with repetitive sequences or contamination. If I saw this, I would be very worried about the quality of my count matrix and how accurate the quantification of gene expression was, and I would not do downstream analyses.


*2. 15% of reads have identical sequences*

I am very concerned about having a high percentage of identical reads because it could indicate PCR duplication or contamination and could skew the gene expression quantification, ultimately influencing our interpretation of differential expression results downstream. 

*3. Average PHRED score < 20 in the last 10 bases*

If I saw a trend of low PHRED scores at the end of reads, I'd be concerned about the sequencing quality and how errors in read ends could impact mapping accuracy and variant calling in later analysis. I'd consider more strict trimming for low-quality bases. 

*4. 10% of reads are unmapped after alignment to the appropriate genome*

I do expect some unmapped reads, but 10% unmapped could imply contamination, sequencing errors, or issues with the reference genome. Since I can filter these reads out, I am concerned but not immensely concerened. 
Other notes: Having 90% mapping is amazing, so 10% unmapped reads is not concerning at all. Mapping can be around 70%, so 90% mapping is great.


*5. Non-random nucleotide distribution in the first 6 bases*

I am not as concerened about this as the other things since this is common (biases introduced during random priming or fragmentation). I would simply trim the first few bases can help correct for this issue.

*6. Unequal Read Lengths*

If I saw that the read lengths were highly variable, I'd think it suggests sequencing errors. But, modern aligners can often handle variable-length reads, making this less concerning compared to the issues I ranked higher.
Other notes: When you get sequencing data, it should be the same. Unequal read lengths mean something wrong happened in sequencing.

*7. Nucleotide frequencies of A, C, T, and G are not equal over the entire read*

I would expect not to not be equal, so this is not concerning. While extreme imbalances could indicate technical issues, I would say this is usually not a major concern.

*8. Unequal number of forward and reverse reads*

This is not concerning as long as the unqueal number is not extreme and both pairs are present. While paired-end sequencing typically produces equal numbers of forward and reverse reads, minor differences can occur due to things like sequencing errors.
Other notes: This means there is some error with sequencing core, we expect to have the same number of reads since they are paired. We cannot fix this with trimming.




# Methods

**Source of data**
The bulk RNAseq data samples were the TYK2 KO at developmental Stage 5 vs the WT at developmental Stage 5 in pancreatic b-cells and can be downloaded from GEO accession (GSE190725).

**Quality Control**
Initial quality control was performed using FastQC v0.12.0. 

**Alignment and Counting**
Reads were aligned to the Gencode hg38 reference genome (GRCh38) using STAR v2.7.11b with default parameters. Gene counts were generated from alignments using VERSE v1.0.5 with default parameters and the gencode hg38 GTF for the primary assembly.

**Filtering and Differential Expression**
Counts were combined and filtered to remove genes that with a total count across all samples less than 10. Normalization and differential expression analysis was performed with DESeq2 v1.44.0 using default parameters comparing the TYK2 KO at Stage 5 vs the WT at Stage 5.




